WordPress/MediaEditor/MediaEditor.swift

```diff
@@ -1,103 +1,67 @@
+    var onFinishEditing: ((UIImage, [MediaEditorOperation]) -> ())?
 
-    public var doneTextButton: UIButton {
-        return cropViewController.toolbar.doneTextButton
-    }
-
-    public var cancelTextButton: UIButton {
-        return cropViewController.toolbar.cancelTextButton
-    }
-
-    public var resetButton: UIButton {
-        return cropViewController.toolbar.resetButton
-    }
-
-    public var doneIconButton: UIButton {
-        return cropViewController.toolbar.doneIconButton
-    }
+    var onCancel: (() -> ())?
 
-    public var cancelIconButton: UIButton {
-        return cropViewController.toolbar.cancelIconButton
-    }
-
-    public var rotateClockwiseButton: UIButton? {
-        return cropViewController.toolbar.rotateClockwiseButton
-    }
-
-    public var rotateCounterclockwiseButton: UIButton? {
-        return cropViewController.toolbar.rotateCounterclockwiseButton
-    }
-
-    public var rotateCounterclockwiseButtonHidden: Bool {
-        get {
-            return cropViewController.toolbar.rotateCounterclockwiseButtonHidden
-        }
+    private(set) var currentCapability: MediaEditorCapability?
 
-        set {
-            cropViewController.toolbar.rotateCounterclockwiseButtonHidden = newValue
+    public var styles: MediaEditorStyles = [:] {
+        didSet {
+            currentCapability?.apply(styles: styles)
         }
     }
 
-    public var rotateClockwiseButtonHidden: Bool {
-        get {
-            return cropViewController.toolbar.rotateClockwiseButtonHidden
-        }
-
-        set {
-            cropViewController.toolbar.rotateClockwiseButtonHidden = newValue
-        }
+    init(_ image: UIImage) {
+        self.image = image
+        hub = MediaEditorHub()
+        super.init(rootViewController: hub)
+        presentIfSingleImageAndCapability()
+        modalTransitionStyle = .crossDissolve
+        modalPresentationStyle = .fullScreen
+        navigationBar.isHidden = true
     }
 
-    public init(cropViewControllerFactory: @escaping (UIImage) -> TOCropViewController = TOCropViewController.init, image: UIImage) {
-        self.cropViewControllerFactory = cropViewControllerFactory
-        self.image = image
-        super.init()
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
     }
 
     public func edit(from viewController: UIViewController? = nil, onFinishEditing: @escaping (UIImage, [MediaEditorOperation]) -> (), onCancel: (() -> ())? = nil) {
         self.onFinishEditing = onFinishEditing
         self.onCancel = onCancel
-        cropViewController.delegate = self
-        viewController?.present(cropViewController, animated: true)
+        viewController?.present(self, animated: true)
     }
 
-    public func dismiss(animated: Bool, completion: (() -> ())? = nil) {
-        cropViewController.dismiss(animated: animated, completion: completion)
-    }
+    func presentIfSingleImageAndCapability() {
```

Comment:
Why are we having this check?

Reply:
Good question! That doesn't make sense so far, but it will!

If we have a single image and a single capability, we will present the capability right away. If not, we will display the Hub.

A few examples:

* Single image and single capability: present the capability
* Multiple images and single capability: present the Hub (the hub will display all the images and the user can choose which one he wants to manipulate)
* Single image and multiple capabilities: present the Hub (the user can then select which capability he want to use: Basics, Brightness, Filter or anything else)
* Multiple images and capabilities: same as above

Does that make sense? I should've avoided including this logic now. I'm sorry about that. But it will be very important, I promise!

Reply:
>But it will be very important, I promise!

Hehe. I believe you 😄